File: 401 to 500/Python/455_assign-cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        res , i , j = 0,len(g)-1,len(s)-1
        while i>-1 and j>-1:
            if g[i] <= s[j]:
                res += 1
                i -= 1
                j -= 1
            else:
                i -= 1
        return res

File: 401 to 500/C++/455_assign-cookies.cpp
#include<bits/stdc++.h>
using namespace std;
#define endl '\n'
#define int long long
const int mod = 1e9+7;

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        if(s.size()==0 || g.size()==0) return 0;
        sort(s.begin(),s.end());
        sort(g.begin(),g.end());
        int res = 0, j=s.size()-1, i=g.size()-1;
        while(j>=0 && i>=0){
            if(g[i]<=s[j]){
                res++;
                j--;
                i--;
            }else i--;
        }
        return res;
    }
};

signed main(){
    ios_base::sync_with_stdio(false);
    cin.tie(0);cout.tie(0);
    Solution obj;
}